libdnf/module/ModulePackage.cpp

```diff
@@ -85,14 +85,22 @@ static std::pair<std::string, std::string> parsePlatform(const std::string & pla
     return make_pair(platform.substr(0,index), platform.substr(index+1));
     }
 
-ModulePackage::~ModulePackage() = default;
+ModulePackage::~ModulePackage()
+{
+    if (mdStream) {
```

Comment:
Sorry about not mentioning this the first time, but it's better to check pointers explicitly (for clarity of coding style, it works the same, of course): `if (mdStream != nullptr)`

There's a lot of those throughout the PR...